Python-Snowflake jobs

Installing snowflake python connector
pip install --upgrade snowflake-connector-python  (for Python 2)
pip3 install --upgrade snowflake-connector-python
Urllib error

  File "/anaconda3/lib/python3.6/site-packages/pip/_internal/download.py", line 480, in path_to_url
    url = urllib_parse.urljoin('file:', urllib_request.pathname2url(path))
  File "/anaconda3/lib/python3.6/urllib/request.py", line 1739, in pathname2url
    return quote(pathname)
NameError: name 'quote' is not defined
Solution: re-install conda and anaconda
(I had a doctored version urrllib)
attempted to install connector within a virtual environment, 
“conda update anaconda” ran but didn’t help
Ran in my home environment:
conda update conda    (updating the package manager, includes new copy of urllib3)
conda update anaconda  (updating update the meta-package)
Tried 
pip3 install --upgrade snowflake-connector-python
yay, snowflake connector installed correctly!


Establish secure snowflake connection
From any text editor, create a file named validate.py with contents: (replace password)
#!/usr/bin/env python
import snowflake.connector

# Gets the version
ctx = snowflake.connector.connect(
   user='resyanalytics',
   password='yourSnowflakePassword',
   account='yk58234.us-east-1'
   )
cs = ctx.cursor()
try:
   cs.execute("SELECT current_version()")
   one_row = cs.fetchone()
   print(one_row[0])
finally:
   cs.close()
ctx.close()

From Terminal, run 
>> python validate.py

If everything runs fine, you should get back the connector version number. 
 
Gateway Timeout error
File "/anaconda3/lib/python3.6/site-packages/snowflake/connector/errors.py", line 73, in default_errorhandler
    done_format_msg=errorvalue.get(u'done_format_msg'))
snowflake.connector.errors.GatewayTimeoutError: HTTP 504: Gateway Timeout

Solution: Disable MFA
Disabled MFA inside Snowflake using:
ALTER USER RESYANALYTICS SET DISABLE_MFA = TRUE;

I should figure out how to make it run with MFA. and with key-pair authentication. 
https://docs.snowflake.net/manuals/user-guide/python-connector-example.html#using-key-pair-authentication


